### PR TITLE
remove soft-deletes

### PR DESCRIPTION
**Pull request by kian99**
## Description

This PR removes the gorm.Model type from ~~most of our db model structs~~ groups to avoid the issue of soft-deletes. Removing it from all models break several tests (it seems because we ignore gorm.Model in these tests).

Fixes [CSS-11335](https://warthogs.atlassian.net/browse/CSS-11335)

[CSS-11335]: https://warthogs.atlassian.net/browse/CSS-11335?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by SimoneDutto — COMMENTED** [review]
i am curious, how is this not requiring a db migration (gorm magic)? Are we fine with removing the soft delete from controllers? Killing a controller on juju can fail and it is async, right?

**Comment by kian99** [comment]
> i am curious, how is this not requiring a db migration (gorm magic)? Are we fine with removing the soft delete from controllers? Killing a controller on juju can fail and it is async, right?

Ah good point, we do need a migration to remove the column. Though it should work fine without the migration because Gorm will just ignore the field.
As for removing soft delete from controllers, that was already done previously, this PR just removes the soft-delete from `CloudRegionControllerPriority` and `ControllerConfig` controller related tables (among others). When you remove a Juju controller from JIMM you basically just detach it.

Edit: But I've edited this now to only remove soft delete for groups. See the description.

File: internal/dbmodel/group.go [hunk 1]
```diff
@@ -6,15 +6,17 @@ import (
...
 
 	apiparams "github.com/canonical/jimm/v3/pkg/api/params"
 	jimmnames "github.com/canonical/jimm/v3/pkg/names"
 )
 
 // A GroupEntry holds information about a user group.
 type GroupEntry struct {
-	gorm.Model
+	// Note this doesn't use the standard gorm.Model to avoid soft-deletes.
+	ID        uint `gorm:"primarykey"`
```
Comment (SimoneDutto) [inline internal/dbmodel/group.go:17]: i could be wrong, but when creating `GroupEntry` this ID is always set to 0. Right?
Or the form struct tag does something to it?

**Review by SimoneDutto — APPROVED** [review]

**Review by SimoneDutto — COMMENTED** [review]
wrong approve

File: internal/dbmodel/group.go (hunk 1 above)
Comment (kian99) [reply internal/dbmodel/group.go:17]: The default value is 0 but gorm will assign the database ID to it. (I hope that answers your question?)

Comment (SimoneDutto) [reply internal/dbmodel/group.go:17]: yes, i was worried we were using that ID for something in the tests. But I've checked and it is alright

**Review by ale8k — APPROVED** [review]
lgtm

**Pull request merged**